File: LLM-Translations/GPT-4/avatar/Java/Python/atcoder_AGC019_A.py
import sys
from math import *

def main():
    Q, H, S, D, N = map(int, sys.stdin.readline().split())
    onePrice = min(min(Q * 4, H * 2), S)
    if onePrice <= D / 2:
        print(N * onePrice)
    else:
        print((N // 2) * D + (N % 2) * onePrice)

main()

File: LLM-Translations/GPT-4/avatar/Java/Go/atcoder_AGC019_A.go
package main

import (
	"fmt"
	"math"
	"os"
	"strconv"
)

func main() {
	var Q, H, S, D, N float64

	Q, _ = strconv.ParseFloat(os.Args[1], 64)
	H, _ = strconv.ParseFloat(os.Args[2], 64)
	S, _ = strconv.ParseFloat(os.Args[3], 64)
	D, _ = strconv.ParseFloat(os.Args[4], 64)
	N, _ = strconv.ParseFloat(os.Args[5], 64)

	onePrice := math.Min(math.Min(Q*4, H*2), S)

	if onePrice <= D/2 {
		fmt.Println(N * onePrice)
	} else {
		fmt.Println((N/2)*D + math.Mod(N, 2)*onePrice)
	}
}

// //End of Code.
